.github/workflows/CI.yml

```diff
@@ -67,8 +67,10 @@ jobs:
         run: julia --project .github/set_ci_preferences.jl "${{ matrix.provider }}"
       - uses: julia-actions/julia-runtest@v1
       - uses: julia-actions/julia-processcoverage@v1
-      - uses: codecov/codecov-action@v3
+      - uses: codecov/codecov-action@v4
         with:
+          fail_ci_if_error: true
```

Comment:
Otherwise codecov failures are silent, which eg caused many packages to upgrade to v4 of the action without noticing that it actually broke codecov integration since tokenless uploads are not supported anymore.

Reply:
Well, now it's failing for no apparent good reason, making all CI red when the tests are actually failing.

Reply:
AFAICT it is failing for a good reason - the token is missing. I'm pretty sure it actually exists though (tests passed in this PR!) but that the problem in the linked run is that dependabot prs don't have access to secrets. Typically retriggering CI (eg by closing and reopening the PR) fixes these problems with insufficient dependabot permissions.

Reply:
>  it is failing for a good reason - the token is missing

No, it's not, it has been there for two months, it has worked so far and it started failing only now

![image](https://github.com/JuliaMath/FFTW.jl/assets/765740/a46f13c3-b6d6-46a7-986e-562b035ff825)

Reply:
I meant: In the run you pointed to - there it was missing because runs triggered by dependabot PRs do not have access to secrets. As I said, I was certain that the token is there in principle. As suspected, closing and reopening the dependabot PR fixed the CI issues.